app/models/user.rb

```diff
@@ -16,9 +18,9 @@ class User < ActiveRecord::Base
   attr_accessible *(ACCESSIBLE_ATTRIBUTES + [:admin]), :as => :admin
 
   validates_presence_of :username
-  validates_uniqueness_of :username
+  validates :username, uniqueness: { case_sensitive: false }
```

Comment:
Ah, yes, I completely forgot about PostgreSQL.  Maybe we should create a function index for `lower(username)` instead on PostgreSQL, or perhaps change the column type to citext. (Does AR support it?)